src/main/scala/com/codacy/api/client/CodacyClient.scala

```diff
@@ -103,9 +105,13 @@ class CodacyClient(
   }
 
   private def parseJson(input: String): RequestResponse[JsValue] = {
-    val json = Json.parse(input)
-    json
-      .validate[ErrorJson]
-      .fold(_ => SuccessfulResponse(json), apiError => FailedResponse(s"API Error: ${apiError.error}"))
+    Try(Json.parse(input)) match {
+      case Success(json) =>
+        json
+          .validate[ErrorJson]
+          .fold(_ => SuccessfulResponse(json), apiError => FailedResponse(s"API Error: ${apiError.error}"))
+      case Failure(exception) =>
+        FailedResponse(s"Failed to parse API response $input as JSON with message ${exception.getMessage}")
```

Comment:
What do you think about returning a more generic error to the user and logging the full thing instead of putting on the message?

Reply:
This client does not have any dependency on logging, that's why I was avoiding to log. I think that in the case of the coverage reporter, it will log failed requests, but other users of the client might not.

Reply:
I really think having this mega messages without types is aweful since all errors are the same.

Reply:
I added a dependency on `scala-logging`.